File: source/com/barrybecker4/puzzle/shuffle/Deck.java
// Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.shuffle;

/**
 * @author Barry Becker
 */
public interface Deck {

    int size();
    int get(int i);
    long shuffleUntilSorted(int iCut);
    void doPerfectShuffle(int iCut);
}

File: scala-source/com/barrybecker4/puzzle/shuffle/Deck.scala
// Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.shuffle

/**
  * @author Barry Becker
  */
trait Deck {

  def size: Int

  def get(i: Int): Int

  def shuffleUntilSorted(iCut: Int): Long

  def doPerfectShuffle(iCut: Int): Unit
}
